src/providers/postgres/qgspgnewconnection.cpp

```diff
@@ -249,7 +249,7 @@ void QgsPgNewConnection::showHelp()
 void QgsPgNewConnection::updateOkButtonState()
 {
   bool enabled = !txtName->text().isEmpty() && (
-                   ( !txtService->text().isEmpty() && !txtDatabase->text().isEmpty() ) ||
+                   ( !txtService->text().isEmpty() && txtDatabase->text().isEmpty() ) ||
```

Comment:
If database is not required we can skip checking `txtDatabase` completely.

Reply:
Yes @alexbruy, you could skip the `txtDatabase` check. 

I would vote to keep the condition as it is, instead of a "don't care" about the database name when the users uses libpq services. It can be misleading to see in the documentation or in web tutorials both the service and the database filled even if we don't care about the database name.

With this condition in place, the user knows that it makes no sense to enter a database name. Since this database name is not used by libqp, we force the user to keep it empty.

But I don't have a strong opinion on this. Dropping the condition as you suggested is our current behaviour. What do you prefer?

Reply:
As *all* parameters saved to the settings, even if they are not used, I think it is safe to drop database name from the condition.

Reply:
Done!